apps/api/src/account/account.service.ts

```diff
@@ -1,13 +1,30 @@
 import { Injectable } from '@nestjs/common'
 import { Person } from '@prisma/client'
 import { UpdatePersonDto } from '../person/dto/update-person.dto'
-import { PrismaService } from '../prisma/prisma.service'
+
+import { KeycloakTokenParsed } from '../auth/keycloak'
+import { AuthService } from '../auth/auth.service'
+import CredentialRepresentation from '@keycloak/keycloak-admin-client/lib/defs/credentialRepresentation'
 
 @Injectable()
 export class AccountService {
-  constructor(private prismaService: PrismaService) {}
+  constructor(private authService: AuthService) {}
+
+  async updateUserProfile(
+    user: KeycloakTokenParsed,
+    data: UpdatePersonDto,
+  ): Promise<Person | null> {
+    return await this.authService.updateUser(user.sub as string, data)
```

Comment:
let's make user.sub mandatory field in KeycloakTokenParsed type, so that we don't need to cast it like this

Reply:
It is used in some other controllers, shall I change it there as well?

Reply:
yes, let's be consistent